lisp/ess-mode.el

```diff
@@ -990,6 +990,29 @@ generate the source buffer."
                   (down-list 1)
                 (error nil)))))))
 
+(defun ess-define-runner (ver dialect)
```

Comment:
The defun+eval thing is nice because defun will eval its argument. Trying to do it with a macro ended up with evalling the arguments anyway, which felt (and looked) messy.

Reply:
I don't think I explained myself well here. Basically, defun+eval lets us have one function that only has one side effect --- it defines another function. That's useful since then we can just use `mapc` over the `versions` variable later on (and `mapc` is specifically for functions that just produce side effects).